evap/student/tests/test_views.py

```diff
@@ -18,63 +19,126 @@ def setUp(self):
+                         "3")
+
+        self.assertEqual(form[question_id(self.contribution1, self.contributor_questionnaire, self.contributor_text_question)].value,
+                         "some other text")
+        self.assertEqual(form[question_id(self.contribution1, self.contributor_questionnaire, self.contributor_likert_question)].value,
+                         "4")
+
+        self.assertEqual(form[question_id(self.contribution2, self.contributor_questionnaire, self.contributor_text_question)].value,
+                         "some more text")
+
+    def test_answer(self):
+        page = self.get_assert_200(self.vote_url(), user=self.voting_user1.username)
         form = page.forms["student-vote-form"]
-        self.assertEqual(form["question_17_2_3"].value, "some text")
-        self.assertEqual(form["question_17_2_4"].value, "1")
-        self.assertEqual(form["question_17_2_5"].value, "6")
-        self.assertEqual(form["question_18_1_1"].value, "some other text")
-        self.assertEqual(form["question_18_1_2"].value, "1")
-        self.assertEqual(form["question_19_1_1"].value, "some more text")
-        self.assertEqual(form["question_19_1_2"].value, "1")
-        self.assertEqual(form["question_20_1_1"].value, "and the last text")
-        form["question_20_1_2"] = 1  # give missing answer
+        self.fill_complete_form(form)
         form.submit()
 
-        self.get_assert_403("/student/vote/5", user="lazy.student")
+        page = self.get_assert_200(self.vote_url(), user=self.voting_user2.username)
+        form = page.forms["student-vote-form"]
+        self.fill_complete_form(form)
+        form.submit()
 
+        self.assertEqual(len(TextAnswer.objects.all()), 6)
+        self.assertEqual(len(RatingAnswerCounter.objects.all()), 4)
 
-    def test_user_cannot_vote_for_themselves(self):
-        contributor1 = mommy.make(UserProfile)
-        contributor2 = mommy.make(UserProfile)
-        student = mommy.make(UserProfile)
+        self.assertEqual(RatingAnswerCounter.objects.filter(question=self.general_likert_question).count(), 1)
+        self.assertEqual(RatingAnswerCounter.objects.get(question=self.general_likert_question).answer, 1)
+
+        self.assertEqual(RatingAnswerCounter.objects.filter(question=self.general_grade_question).count(), 1)
+        self.assertEqual(RatingAnswerCounter.objects.get(question=self.general_grade_question).answer, 3)
+
+        self.assertEqual(RatingAnswerCounter.objects.filter(question=self.contributor_likert_question).count(), 2)
+        self.assertEqual(RatingAnswerCounter.objects.filter(question=self.contributor_likert_question)[0].answer, 4)
+        self.assertEqual(RatingAnswerCounter.objects.filter(question=self.contributor_likert_question)[1].answer, 2)
+
+        self.assertEqual(TextAnswer.objects.filter(question=self.general_text_question).count(), 2)
+        self.assertEqual(TextAnswer.objects.filter(question=self.contributor_text_question).count(), 4)
```

Comment:
can you check that the answers belong to the right contribution? also, can you check the contents of some (not necessarily all text answers?